### DISCUSSION TITLE
Some ideas for fluid wallpaper

### DISCUSSION THREAD

**Discussion by SirTwsted** [rocksdanister/lively#1517] [General]
Hi,

I just installed the Lively Wallpaper,  And I am using the Fluids wallpaper.
But I have 1 HUGE problem with this wallpaper.

There is no automated liquid.  It only interacts with my mouse.
I want the wallpaper to do liquid on its own.
Also,  There needs to be more options besides just (Random Splats)  The random splashs is too flashy.

![Screenshot_283](https://user-images.githubusercontent.com/15247713/210240812-6361de66-5d6a-40ce-9166-bf659b9b4453.jpg)

Liquid is suppose to flow in many colors and it should also allow me to chose what colors,
and how many POINTS do I want the liquids to start as well as Fade-Ins for those points for those liquids fluids.

I have Fluid and Magic fluids on my cell phone by a different developer.
And they have a setting that does the fluid animation all my itself, and you can set as many points as you want.

I want that for my desktop in this fluids wallpaper please.
I should be able to choose a shape,  the colors and how many points I want.
As I have (3) 32-inch monitors at 2160x1440p   I should be able to see liquid all over my 3 monitor screens.

But at the current time, I don't.  It only shows when I move the mouse.
Which defeats the purpose of wallpaper.

Also, I would would like to have the option set Background Image as a Video File,  for those of us that have a 30 second video loop,
of something nice and be able to also have the liquid fluid on top, instead of only a flat image as the background..

![image-background-fluid](https://user-images.githubusercontent.com/15247713/210240606-899e1902-061c-43ac-bcbd-846068bcb35b.jpg)

**Comment by SirTwsted** [comment]
Also I forgot to mention, 
I tried to add my own 10 second loop video as background.  and the image is 2560x1440 , the same as my 3 monitors.
But for some reason your software takes my video andstretches it very weird.
I tried everything to try to fix it and it wont fix. 
I even went to the settings window to try to figure out whats going on and nothing there fixes it,.
To place that video on 3 monitors at its proper size. 

![Screenshot_298](https://user-images.githubusercontent.com/15247713/210241558-9bc0cb56-1e54-453c-b6ef-c0652b4cbb18.jpg)


I cant seem to get this software to place my 10-sec video clip I want to use as a background wallpaper
to its proper size.
Again, my monitor is 2560x1440p  and the video is 2560x1440p  and I want the video to be used on all 3 monitors.
To be used behind the fluids affects together.  The Uniform fixes the video to its proper size but it wont place it on my 3 monitors.


:)  thanks.

**Comment by SirTwsted** [comment]
This is an example of how it SHOULD look on 3 minitors.
This is me using my mouse and clicking around to get the effect I want.  This program should be able to do this
without me using my mouse and clicking.  By me selecting how many Fluid points I want activated and 
Pressure of those points when it it starts.  Without using Splash pop because it SHOULD be fluid aka Liquid.

Anyways, here is a example of what I mean.
This is 3 monitors screenshot. - Dont mind the black icon on the desktop I forgot to delete it.

![Screenshot_299](https://user-images.githubusercontent.com/15247713/210242681-050d9340-ab1f-4af2-a239-2599d8d36d0f.jpg)


:)

**Comment by SirTwsted** [comment]
Did I do something wrong to get my post locked ???  
I am a bit confused  :(

**Reply by rocksdanister** [reply]
Nah, just converted the [issue](https://github.com/rocksdanister/lively/issues) to [discussion](https://github.com/rocksdanister/lively/discussions) since its about the wallpaper and not the software.

**Comment by HaLo2FrEeEk** [comment]
Check out [this wallpaper animation](https://github.com/HaLo2FrEeEk/Starfield-wallpaper-for-Lively) that I wrote. It's not fluid like the WebGL one, but I think it's quite nice and it just does its own thing in the background.

**Comment by pitfiend** [comment]
This is awesome... but needs to have some way to choose color palettes. Having random colors is just weird in my opinion and impose a limitation. I would like to have a blue fire palette, for example. Is this possible?

**Reply by rocksdanister** [reply]
possible

https://github.com/rocksdanister/WebGL-Fluid-Simulation/blob/453d97445ca186d1d8195d2d8dfdc4a9bc272bce/js/script.js#L1661-L1667

**Reply by rocksdanister** [reply]
Video background, Custom color and overlay:
https://github.com/rocksdanister/WebGL-Fluid-Simulation/releases/tag/v6
